src/client/components/localisation/network.ts

```diff
@@ -0,0 +1,45 @@
+import { action } from 'mobx'
+import { message } from '../../../shared/proto/messages'
+import { Network } from '../../network/network'
+import { LocalisationModel } from './model'
+import Sensors = message.input.Sensors
+import { inject } from 'inversify'
+import { injectable } from 'inversify'
+
+@injectable()
+export class LocalisationNetwork {
+  public constructor(@inject(Network) private network: Network,
+                     @inject(LocalisationModel) private model: LocalisationModel) {
+    this.network.on(Sensors, this.onSensors)
+  }
+
+  public destroy() {
+    this.network.off()
+  }
+
+  @action
+  private onSensors = (sensors: Sensors) => {
+    // TODO (Annable): Remove hardcoded values.
```

Comment:
Ah, not quite. It's currently hardcoded to use `this.model.robots[0]`, instead of actually knowing about the connected robots (next PR).